regexp.go

```diff
@@ -109,6 +109,12 @@ func newRouteRegexp(tpl string, matchHost, matchPrefix, matchQuery, strictSlash,
 	if errCompile != nil {
 		return nil, errCompile
 	}
+
+	// Check for capturing groups which used to work in older versions
+	if reg.NumSubexp() != len(idxs)/2 {
+		panic("Regexp appears to be using capturing groups. To fix, see: https://github.com/gorilla/mux/issues/200")
```

Comment:
We shouldn't link to the issue. We should instead show the failing route - "route %q contains capture groups in its regexp. Only non-capturing groups are accepted: e.g. (?:pattern) instead of (pattern)"

(or an approximation of that)